test/e2e/data/e2e_conf.yaml

```diff
@@ -58,8 +58,8 @@ providers:
 - name: kubeadm
   type: ControlPlaneProvider
   versions:
-  - name: v1.0.0
-    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.0/control-plane-components.yaml"
+  - name: v1.0.4
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.4/control-plane-components.yaml"
     type: url
     files:
```

Comment:
You have to change the CAPO version below to v0.5.99. Essentially that is a "dev version" which matches the latest release series of your metadata.yaml

Reply:
You should also drop the 1.0 series in `../data/shared/v1beta1_provider/metadata.yaml` (like in the top-level metadata.yaml)